### PR TITLE
feat: full refactor

### PR DESCRIPTION
**Pull request by Pantani**

### CODE REVIEW

File: .github/workflows/release-homebrew.yml
Comment (julienrbrt) [inline .github/workflows/release-homebrew.yml:1]: we don't need all those workflows

Comment (Pantani) [reply .github/workflows/release-homebrew.yml:1]: the idea is to release gex in a homebrew formula

Comment (julienrbrt) [reply .github/workflows/release-homebrew.yml:1]: Oh really, cool!

File: .github/workflows/release-homebrew.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - uses: mislav/bump-homebrew-formula-action@v3
+        with:
+          formula-name: gex
+          formula-path: Formula/i/gex.rb
+          homebrew-tap: Homebrew/homebrew-core
+          base-branch: master
+          download-url: https://github.com/ignite/gex/archive/refs/tags/${{ steps.extract-version.outputs.tag-name }}.tar.gz
+        env:
+          # the personal access token should have "repo" & "workflow" scopes
+          COMMITTER_TOKEN: ${{ secrets.COMMITTER_TOKEN }}
```
Comment (julienrbrt) [inline .github/workflows/release-homebrew.yml]: cc @ilgooz, you'll need to add the same token from ignite/cli here

Comment (ilgooz) [reply .github/workflows/release-homebrew.yml]: is it still needed?

**Pull request merged**